src/features/defi/providers/univ2/hooks/useUniV2LiquidityPool.ts

```diff
@@ -186,40 +286,108 @@ export const useUniV2LiquidityPool = ({
-      asset1ContractAddress,
+      asset0,
+      assetId0OrWeth,
+      assetId1OrWeth,
+      makeAddLiquidityData,
       skip,
       supportedEvmChainIds,
       uniswapRouterContract,
       wallet,
+      weth,
     ],
   )
 
+  const makeRemoveLiquidityData = useCallback(
+    ({
+      asset0ContractAddress,
+      asset1ContractAddress,
+      lpAmount,
+      asset1Amount,
+      asset0Amount,
+    }: {
+      asset0ContractAddress: string
+      asset1ContractAddress: string
+      lpAmount: string
+      asset1Amount: string
+      asset0Amount: string
+    }) => {
+      if (!uniswapRouterContract) throw new Error('uniswapRouterContract not defined')
+
+      const lpAmountBaseUnit = bnOrZero(lpAmount)
+        .times(bn(10).exponentiatedBy(lpAsset.precision))
+        .toFixed(0)
+      const deadline = Date.now() + 1200000 // 20 minutes from now
+      const to = fromAccountId(accountId).account
+      if ([assetId0OrWeth, assetId1OrWeth].includes(wethAssetId)) {
+        const otherAssetContractAddress =
+          assetId0OrWeth === wethAssetId ? asset1ContractAddress : asset0ContractAddress
+        const otherAsset = assetId0OrWeth === wethAssetId ? asset1 : asset0
+        const ethAmount = assetId0OrWeth === wethAssetId ? asset0Amount : asset1Amount
+        const otherAssetAmount = assetId0OrWeth === wethAssetId ? asset1Amount : asset0Amount
+        return uniswapRouterContract.interface.encodeFunctionData('removeLiquidityETH', [
+          otherAssetContractAddress,
+          lpAmountBaseUnit,
+          calculateSlippageMargin(otherAssetAmount, otherAsset.precision),
+          calculateSlippageMargin(ethAmount, weth.precision),
+          to,
+          deadline,
+        ])
+      }
+
+      return uniswapRouterContract.interface.encodeFunctionData('removeLiquidity', [
+        asset0ContractAddress,
+        asset1ContractAddress,
+        lpAmountBaseUnit,
+        calculateSlippageMargin(asset0Amount, asset0.precision),
+        calculateSlippageMargin(asset1Amount, asset1.precision),
+        to,
+        deadline,
+      ])
+    },
+    [
+      accountId,
+      asset0,
+      asset1,
+      assetId0OrWeth,
+      assetId1OrWeth,
+      lpAsset.precision,
+      uniswapRouterContract,
+      weth.precision,
+    ],
+  )
   const removeLiquidity = useCallback(
-    async (lpAmount: string, asset1Amount: string, asset0Amount: string) => {
+    async ({
+      lpAmount,
+      asset1Amount,
+      asset0Amount,
+    }: {
+      lpAmount: string
+      asset1Amount: string
+      asset0Amount: string
+    }) => {
       try {
         if (skip || !accountId || !isNumber(accountNumber) || !uniswapRouterContract || !wallet)
           return
         const chainAdapterManager = getChainAdapterManager()
         const adapter = chainAdapterManager.get(asset0.chainId) as unknown as ethereum.ChainAdapter
         if (!adapter) throw new Error(`addLiquidityEth: no adapter available for ${asset0.chainId}`)
```

Comment:
```suggestion
        if (!adapter) throw new Error(`removeLiquidity: no adapter available for ${asset0.chainId}`)
```

same existing bug/typo on L404

Reply:
dce433372302f57a76c30d10dd317c427cf444ce / 3d23ba6fe1b21e3f65d757fabc23f26247cd2e33